src/makeInfinityAdder.js

```diff
@@ -4,7 +4,31 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  const adder = function (a) {
+    let sum;
+
+    if (a === undefined) {
+      if (sum === undefined) {
+        sum = 0;
```

Comment:
The variable 'sum' should be declared outside of the 'adder' function to retain its value between calls. Otherwise, it will be reset every time 'adder' is called, which will not allow the function to accumulate the sum of the numbers.

Reply:
In my case, the variable 'sum' works correctly inside the function 'adder' as well. If I declare it outside of the 'adder' function I need to change the whole logic of the code